nexus_motion_planner/src/motion_planner_server.hpp

```diff
@@ -88,6 +95,19 @@ class MotionPlannerServer : public rclcpp_lifecycle::LifecycleNode
   double _workspace_max_y;
   double _workspace_max_z;
 
+  // Motion plan caching
+  std::shared_ptr<nexus::motion_planner::MotionPlanCache> _motion_plan_cache;
+
+  bool _use_motion_plan_cache;
+  bool _use_cached_plans_instead_of_planning;
+  bool _overwrite_worse_plans;
+  std::string _cache_db_plugin;
+  std::string _cache_db_host;
+  int _cache_db_port;
+  double _cache_exact_match_tolerance;
```

Comment:
Instead of three doubles, let's define an enum class for each of these cases. Then have two params: 1) To get the string name of the enum type and match as defined above 2) The value of the tolerance. You could also consider using `std::variant` here and define 3 three different structs with boolean values of tolerance.

Reply:
Any cache will use all three of these tolerance values (exact matches for population, and start and goal for plan fetches.)

More importantly, these values should be configurable by the user.

I don't understand why we would use an enum or variant in that case?
Maybe I'm missing something here?

Reply:
I think enums and variants don't quite make sense here also. Perhaps there is confusion with the naming having both "exact_match" and "tolerance", the fomer suggest a bool and the latter suggest a double.

Reply:
I'll add comments to explain:
ccbf91cd9f98e626bd1ba4a535b5b2aff10eba0c in https://github.com/osrf/nexus/pull/26